src/gcp/resourceManager.ts

```diff
@@ -98,3 +101,42 @@ export async function addServiceAccountToRoles(
 
   return setIamPolicy(projectId, projectPolicy, "bindings");
 }
+
+export async function serviceAccountHasRoles(
+  projectId: string,
+  serviceAccountName: string,
+  roles: string[],
+  skipAccountLookup = false
+): Promise<boolean> {
+  const [{ name: fullServiceAccountName }, projectPolicy] = await Promise.all([
+    skipAccountLookup
+      ? Promise.resolve({ name: serviceAccountName })
+      : getServiceAccount(projectId, serviceAccountName),
+    getIamPolicy(projectId),
+  ]);
+
+  // The way the service account name is formatted in the Policy object
+  // https://cloud.google.com/iam/docs/reference/rest/v1/Policy
+  // serviceAccount:my-project-id@appspot.gserviceaccount.com
+  const memberName = `serviceAccount:${fullServiceAccountName.split("/").pop()}`;
+
+  for (const roleName of roles) {
+    const bindingIndex = findIndex(
+      projectPolicy.bindings,
+      (binding: Binding) => binding.role === roleName
+    );
+
+    if (bindingIndex === -1) {
+      return false;
+    }
+
+    const binding = projectPolicy.bindings[bindingIndex];
```

Comment:
native js makes this easier

```suggestion
    const binding = projectPolicy.bindings.find((b: Binding) => b.role === roleName);
    if (!binding) {
      return false;
    }
```